packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.service.js

```diff
@@ -1,22 +1,111 @@
-export default class publigGatewaysServiceClass {
-  static getIcebergHeaders() {
-    return {
-      headers: {
-        'X-Pagination-Mode': 'CachedObjectList-Pages',
-        'X-Pagination-Size': 50000,
-        Pragma: 'no-cache',
-      },
-    };
-  }
+import { GATEWAY_DEFAULT_REGION } from './public-gateways.constants';
 
+export default class publicGatewaysServiceClass {
   /* @ngInject */
-  constructor(OvhApiCloudProjectNetworkPrivate, $http) {
+  constructor(
+    OvhApiCloudProjectNetworkPrivate,
+    $http,
+    $q,
+    OvhApiCloudProject,
+    PciProjectRegions,
+    ovhManagerRegionService,
+    Poller,
+    OvhApiCloudProjectNetworkPrivateSubnet,
```

Comment:
Better i think:

```suggestion
    $http,
    $q,
    PciProjectRegions,
    ovhManagerRegionService,
    Poller,
    OvhApiCloudProject,
    OvhApiCloudProjectNetworkPrivate,
    OvhApiCloudProjectNetworkPrivateSubnet,
```

Reply:
I removed all but `$http` as we don't need others.